assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -585,6 +585,15 @@ const ExternalActivity: React.FC<PartComponentProps<CapiIframeModel>> = (props)
             typeof changedVar.value === 'number'
           ) {
             changedVar.type = CapiVariableTypes.NUMBER;
+          } else if (
+            changedVar.type === CapiVariableTypes.STRING &&
+            !Number.isNaN(changedVar.value)
+          ) {
+            const parsedValue = parseNumString(changedVar.value);
+            if (typeof parsedValue === 'number') {
```

Comment:
value of 'parsedValue' can be anything so if it's a type number then only we modifying the value and type.